src/main/proto/uri.proto

```diff
@@ -20,20 +20,94 @@
+  // Any additional details needed
+  repeated google.protobuf.Any details = 10;
+}
+
+
+// An Authority represents the deployment location of a specific Software Entity.
+// Data representation of an <b>Authority</b>.<br> An Authority consists of a device,
+// a domain and an address for the device.<br>
+// Device and domain names are used as part of the URI for device and service discovery.
+message UAuthority {
+  bool local = 1;     // True if local authority, false if remote authority
+  
+  // Remote authority information
+  oneof remote {
+    string name = 2;		// domain & device name as a string
+    bytes ip = 3; 			// IPv4 or IPv6 Address
+    bytes id = 4;			  // Unique ID for the device, could be a VIN, SHA 128, or any other identifier
+                        // *NOTE:* MAX length is 255 bytes
+  }
+}
+
+
+// Data representation of an <b> Software Entity - uE</b><br>
+// Software entities are distinguished by using a unique name and id 
+// along with the specific version of the software.
+// An Software Entity is a piece of software deployed somewhere on a uDevice.<br>
+// The Software Entity is used in the source and sink parts of communicating software.<br>
+// A uE that publishes events is a <b>Service</b> role.<br>
+// A uE that consumes events is an <b>Application</b> role.
+message UEntity {
+  string name = 1;    // Name of the entity
+  uint32 id = 2;      // The numeric ID for the uEntity assigned from the name/number registry
+  optional string version = 3; // optional version of the uEntity
+}
+
+
+// A service API - defined in the {@link UEntity} - has Resources and Methods. 
+// Both of these are represented by the UResource message.
+// A uResource represents a resource from a Service such as "door" and an optional
+// specific instance such as "front_left". In addition, it can optionally contain
+// the name of the resource Message type, such as "Door". The Message type matches
+// the protobuf service IDL that defines structured data types.
+// An UResource is something that can be manipulated/controlled/exposed by a service.
+// Resources are unique when prepended with UAuthority that represents the device and
+// UEntity that represents the service. 
+message UResource {
+  string name = 1;      // Name of the resource
+  string instance = 2;  // Instance of the resource
+  optional string message = 3;   // Message type for the resource
+  uint32 id = 4;        // ID that represents the method/topic for the resource
+}
+
+
+// Repeated list of URIs that can be used for batching APIs
+message BatchUUri {
+  repeated UUri uris = 1; // Repeated list of URIs
+}
+
+
 // Uri passed to an RPC call containing the uri, credentials, and any other
 // information that might be required
 message UriRequest {
-  // uProtocol formatted URI
-  string uri = 1;
+  // uProtocol URI serialized in Long format per 
+  // https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/basics/uri.adoc[URI Specifications]
+  UUri uri = 1;
```

Comment:
the name should be source - just uri is very confusing. This is a request, it has a requestor and what we are requesting.
The naming should be explicit

Reply:
should a request not have a sink? am I getting confused here?